### ISSUE TITLE
URGENT: Tables leaked into document data ; Cannot proceed with Annotation Tag Distribution Analysis

### ISSUE THREAD

**Issue by kamilkrukowski** [kamilkrukowski/FinDocNLP#63]
@NC-Zhao When we first talked about the annotated document parser we discussed that \<table\> annotations should be kept separate from all other non-table annotations

However, they are not separated at all -- This is breaking the machine learning model and analysis downstream.
I just found this out after working on figures and documents analyzing the distribution of annotations tags for a solid 4 hours.

I will  have to re-do all of the analysis -- but I cannot proceed until the annotated document parser is fixed. Therefore my contributions are on-hold until this is fixed.

**Comment by kamilkrukowski** [comment]
There should be a  `headless` argument for the parser and if you set it to false then it will open and render a window. Otherwise `save=True` outputs the file (with highlights if requested). `parser._parse_unannotated_text` does need the same fix.

I can run a check on the current pull request — can you pull request the same code fix on the unannotated parser?

**Comment by kamilkrukowski** [comment]
@NC-Zhao  I can confirm that the code does not work at all. It finds '0' table elements, that is none.

I've been chasing this rabbit hole for the entire morning -- I believe I found a patch after a few hours.

```parent.find_element(....)``` returns a ```child``` or list. However, ```child.parent``` returns the tag which found the child in the first place --> Therefore ```found.parent``` always returns the root webdriver and never actually traverses the HTML tree.

The careful approach is to use ```driver.execute_script('return arguments[0].parentNode', child)``` which will return the HTML parent of the child, not the parent of the traversal function find_element that was previously used. 

TLDR: The functions called did different things than expected, neither of us read what the functions did, and the previously the code returned no elements which was quick to catch in testing (but wasn't tested until this morning)

**Comment by kamilkrukowski** [comment]
I am working on a ```_parse_annotated_files``` patch currently -> could you extend to ```unannotated_files``` when you get a chance after I push?

**Comment by NC-Zhao** [comment]
@kamilkrukowski Let me know when you push, I will be start working on that when I recover from covid booster.

**Comment by kamilkrukowski** [comment]
@NC-Zhao pushed

**Comment by kamilkrukowski** [comment]
We will close this when ```unannotated_files``` is added to ```featurize_file```
